### PR TITLE
[Event Hubs] Support for local non-TLS

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to add unofficial support for the forthcoming Event Hubs local development emulator.  This is intended to support internal testing on very early alpha builds.

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs [hunk 1]
```diff
@@ -38,7 +38,10 @@ internal class AmqpConnectionScope : IDisposable
         private const string WebSocketsPathSuffix = "/$servicebus/websocket/";
 
         /// <summary>The URI scheme to apply when using web sockets for service communication.</summary>
-        private const string WebSocketsUriScheme = "wss";
+        private const string WebSocketsSecureUriScheme = "wss";
+
+        /// <summary>The URI scheme to apply when using web sockets for service communication.</summary>
+        private const string WebSocketsInsecureUriScheme = "wss";
```
Comment (JoshLove-msft) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs:44]: hmm... are these meant to be the same value?

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs [hunk 2]
```diff
@@ -232,8 +235,8 @@ public bool IsDisposed
                                    IWebProxy proxy,
                                    TimeSpan idleTimeout,
                                    string identifier = default,
-                                   int sendBufferSizeBytes = AmqpConstants.TransportBufferSize,
-                                   int receiveBufferSizeBytes = AmqpConstants.TransportBufferSize,
+                                   int sendBufferSizeBytes = -1,
```
Comment (JoshLove-msft) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs:238]: Why remove the use of the constant?

File: sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConnectionOptions.cs [hunk 3]
```diff
@@ -148,5 +158,19 @@ public TimeSpan ConnectionIdleTimeout
...
+
+        /// <summary>
+        ///   Creates a new copy of the current <see cref="EventHubConnectionOptions" />, cloning its attributes into a new instance.
+        /// </summary>
+        ///
+        /// <returns>A new copy of <see cref="EventHubConnectionOptions" />.</returns>
+        ///
+        internal EventHubConnectionOptions Clone()
+        {
+            var clone = EventHubConnectionOptionsExtensions.Clone(this);
```
Comment (JoshLove-msft) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConnectionOptions.cs:170]: Can this property be baked into the Clone method?

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs (hunk 2 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs:238]: This should explicitly be -1 to force the platform defaults to be used.  The constant is the old T0 8192 value that doesn't perform well on linux.   We made the change to the options a little while back but missed the defaults here, which really just impact tests.

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConnectionOptions.cs (hunk 3 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConnectionOptions.cs:170]: Not without making it public.   I'm still tooling around with the structure of things and I think I can pull this into the client directly, so this is likely to go away.

Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConnectionOptions.cs]: I was able to limit the scope of where this was used enough that it doesn't have to flow with the options.   This is now just passed into the `AmqpClient` constructor.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConnectionOptions.cs [hunk 4]
```diff
@@ -7,6 +7,7 @@
 using System.Net.Security;
 using System.Net.Sockets;
 using Azure.Core;
+using Azure.Messaging.EventHubs.Core;
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConnectionOptions.cs:10]:
```suggestion
```

This should be a leftover from pre-refactoring.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubsConnectionStringProperties.cs [hunk 5]
```diff
@@ -2,8 +2,11 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Buffers;
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubsConnectionStringProperties.cs:5]: Double-check usings.   This seems like I forgot to remove unused.

**Pull request merged**